### DISCUSSION TITLE
请问检查音乐可用的的原始数据为什么会是和获取URL类似的结果？

### DISCUSSION THREAD

**Discussion by JackuXL** [Binaryify/NeteaseCloudMusicApi#1564] [Q&A] [answered]
@Binaryify 
代码中，音乐可用性的地址是`/weapi/song/enhance/player/url`，获取URL的地址是`/eapi/song/enhance/player/url`，能否问下这两个地址的来源

**Comment by Binaryify** [comment, marked as answer]
这个接口没啥意义，返回了url就说明可用，只是这个接口一开始是抓网页html，然后正则提取数据，后面才切到现在这个接口url的

**Reply by JackuXL** [reply]
这样啊，谢谢大佬

**Comment by JackuXL** [comment]
@Binaryify 再麻烦问一下，就是weapi eapi分别是哪个平台获取到的，可以替代吗😂

**Reply by Binaryify** [reply]
weapi是网页端的，eapi是客户端的，大部分情况差不多
